sandbox_dns_unix.go

```diff
@@ -313,8 +315,31 @@ func (sb *sandbox) rebuildDNS() error {
 	// external v6 DNS servers has to be listed in resolv.conf
 	dnsList = append(dnsList, resolvconf.GetNameservers(currRC.Content, types.IPv6)...)
 
-	// Resolver returns the options in the format resolv.conf expects
-	dnsOptionsList = append(dnsOptionsList, sb.resolver.ResolverOptions()...)
+	// If the user entered a ndots value > 0 remember that. container's resolv.conf
+	// will only have ndots:0 for embedded DNS server to get unqualified names.
+	resOptions := sb.resolver.ResolverOptions()
+	for i, option := range dnsOptionsList {
+		if strings.Contains(option, "ndots") {
+			parts := strings.Split(option, ":")
+			if len(parts) == 0 || len(parts) > 2 {
+				return fmt.Errorf("invalid ndots option %v", option)
+			}
+			if num, err := strconv.Atoi(parts[1]); err != nil {
+				return fmt.Errorf("invalid ndots option %v", option)
+			} else if num > 0 {
+				sb.ndotsSet = true
+			}
+			for _, resOpt := range resOptions {
+				if strings.Contains(resOpt, "ndots") {
```

Comment:
Yes, `resOptions` is statically defined. But an [earlier comment] (https://github.com/docker/libnetwork/pull/1435#discussion-diff-77909138) was to make the logic generic.

Not sure what you mean by cleanly break out of the outer loop. The processing is done once we hit the `ndots` in the inner loops. So the current form is the idiomatic usage of `goto`; ie: break out of nested loops without extra flags.